.golangci.yml

```diff
@@ -0,0 +1,128 @@
+# Copyright 2023 The Authors (see AUTHORS file)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+run:
+  # default: '1m'
+  timeout: '5m'
+
+  # default: []
+  build-tags:
+  - 'all'
+
+  # default: []
+  skip-dirs:
+  - 'internal/pb'
+
+  # default: true
+  skip-dirs-use-default: false
+
+  # default: ''
+  modules-download-mode: 'readonly'
+
+  # default: false
+  allow-parallel-runners: true
+
+linters:
+  enable:
+  - 'asasalint'
+  - 'asciicheck'
+  - 'bidichk'
+  - 'bodyclose'
+  - 'containedctx'
+  - 'depguard'
+  - 'dupword'
+  - 'durationcheck'
+  - 'errcheck'
+  - 'errchkjson'
+  - 'errname'
+  - 'errorlint'
+  - 'execinquery'
+  - 'exhaustive'
+  - 'exportloopref'
+  - 'forcetypeassert'
+  - 'gci'
+  - 'godot'
+  - 'gofmt'
+  - 'gofumpt'
+  - 'goheader'
+  - 'goimports'
+  - 'gomodguard'
+  - 'goprintffuncname'
+  - 'gosec'
+  - 'gosimple'
+  - 'govet'
+  - 'importas'
+  - 'ineffassign'
+  - 'loggercheck'
+  - 'makezero'
+  - 'misspell'
+  - 'noctx'
+  - 'nolintlint'
+  - 'nosprintfhostport'
+  - 'paralleltest'
+  - 'prealloc'
+  - 'predeclared'
+  - 'revive'
+  - 'rowserrcheck'
+  - 'sqlclosecheck'
+  - 'staticcheck'
+  - 'stylecheck'
+  - 'tenv'
+  - 'thelper'
+  - 'tparallel'
+  - 'typecheck'
+  - 'unconvert'
+  - 'unused'
+  - 'whitespace'
+  - 'wrapcheck'
+
+issues:
+  # default: []
+  exclude:
+  - '^G102:' # gosec: we have to bind to all ifaces in Cloud Run services
+
+  # default: 50
+  max-issues-per-linter: 0
+
+  # default: 3
+  max-same-issues: 0
+
+linters-settings:
+  gofumpt:
+    # default: false
+    extra-rules: true
+
+  goheader:
+    values:
+      regexp:
+        YEAR_AUTHOR: '\d{4} .*'
+    # default: ""
+    template: |-
+      Copyright {{ YEAR_AUTHOR }}
+
+      Licensed under the Apache License, Version 2.0 (the "License");
+      you may not use this file except in compliance with the License.
+      You may obtain a copy of the License at
+
+          http://www.apache.org/licenses/LICENSE-2.0
```

Comment:
When using the [addlicense](https://github.com/google/addlicense) tool i suggested, there are 5 spaces before the apaches license url. I think we should find a tool that uses 4 or just make this 5 spaces and standardize on it.

Reply:
Still a WIP, but I made it regex to match 2+ spaces/tabs